### PR TITLE
Fixed default values for room creation

### PR DESCRIPTION
**Pull request by kriscfoster**
- Setting the room type shouldn't try to change active / externalId values
- Default value for externalId should be null instead of ''
- Issue with default externalId being '' is that it doesn't match the validation regex defined in db. Better just default to null as allowNull is true
- Only require params of room name and roomType to create room

### CODE REVIEW

File: view/rooms/new/FormController.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ class FormController extends React.Component {
     this.getTypes();
   }
 
+  // Gets the room types that are available for use
   getTypes(){
```
Comment (cassiodias) [inline view/rooms/new/FormController.js]: Would be possible to change the method name to getAvailableRoomType and then remove the comment? Comments are really easy to be outdated.

File: view/rooms/new/FormController.js [hunk 2]
```diff
@@ -76,6 +77,7 @@ class FormController extends React.Component {
       });
   }
 
+  // Tries to create a room based on current form.
```
Comment (cassiodias) [inline view/rooms/new/FormController.js]: Isnt this method self-explanatory? Do we need this comment?

File: view/rooms/new/app.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ function getPathVariables(addr){
 
   const paramType = qdata.roomType || '';
   const paramActive = qdata.active !== 'false';
-  const paramExternalId = qdata.externalId || '';
+  const paramExternalId = qdata.externalId || null;
```
Comment (cassiodias) [inline view/rooms/new/app.js]: Is there any expectation or new effect (test) from '' to null?

Comment (kriscfoster) [reply view/rooms/new/app.js]: Thanks @cassiodias for bringing my attention to this. The reason we need to change the default from '' to null is that '' does not pass the regex validation defined for externalId

**Pull request merged**